pkg/migration/5_insert_avz_into_nodepools.go

```diff
@@ -0,0 +1,67 @@
+package migration
+
+import (
+	"fmt"
+
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	v1 "github.com/sapcc/kubernikus/pkg/apis/kubernikus/v1"
+	"github.com/sapcc/kubernikus/pkg/client/openstack"
+)
+
+func InsertAVZIntoNodePools(rawKluster []byte, current *v1.Kluster, client kubernetes.Interface, openstackFactory openstack.SharedOpenstackClientFactory) (err error) {
+	apiSecret, err := client.CoreV1().Secrets(current.Namespace).Get(fmt.Sprintf("%s-secret", current.Name), meta_v1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	secret, err := v1.NewSecret(apiSecret)
```

Comment:
This can be simplified to use `secret, err := util.KlusterSecret(current)`

Reply:
Optimized!